editor/editor_file_dialog.cpp

```diff
@@ -1294,6 +1294,18 @@ void EditorFileDialog::_update_favorites() {
 	favorite->set_pressed(false);
 
 	Vector<String> favorited = EditorSettings::get_singleton()->get_favorites();
+
+	bool fav_changed = false;
+	for (int i = favorited.size() - 1; i >= 0; i--) {
+		if (!dir_access->dir_exists(favorited[i])) {
+			favorited.erase(favorited[i]);
```

Comment:
```suggestion
			favorited.remove_at(i);
```

This avoids element searching. Same in `FileSystemDock`.